github_actions/spec/dependabot/github_actions/file_parser_spec.rb

```diff
@@ -140,6 +140,43 @@
       end
     end
 
+    describe "with reusable workflow" do
+      subject(:dependency) { dependencies.first }
+      let(:workflow_file_fixture_name) { "workflow_reusable.yml" }
+
+      let(:expected_requirements) do
+        [{
+          requirement: nil,
+          groups: [],
+          file: ".github/workflows/workflow.yml",
+          source: {
+            type: "git",
+            url: "https://github.com/actions/checkout",
+            ref: "v2.1.0",
+            branch: nil
+          },
+          metadata: { declaration_string: "actions/checkout/.github/workflows/test.yml@v2.1.0" }
+        }]
+      end
+
+      it "has the right details" do
+        expect(dependency).to be_a(Dependabot::Dependency)
+        expect(dependency.name).to eq("actions/checkout")
+        expect(dependency.version).to eq("2.1.0")
+        expect(dependency.requirements).to eq(expected_requirements)
+      end
+    end
+
+    describe "with empty" do
+      subject(:dependency) { dependencies.first }
+      let(:workflow_file_fixture_name) { "empty.yml" }
+
+      it "has the no dependencies" do
```

Comment:
```suggestion
      it "has no dependencies" do
```

Reply:
No complaints from me, I am not super familiar with specs and am still learning ruby :)

Reply:
It's a great PR, well covered by tests and implementation a very requested feature :) Thanks 😍.

I just noticed that the final diff was a bit bigger than it could, due to the different commits refining things and iteratively getting this to work properly. I pushed a slightly cleaner version that I made while reviewing!